gateway-service/src/test/java/org/zowe/apiml/gateway/conformance/VerificationOnboardServiceTest.java

```diff
@@ -42,20 +48,90 @@ void whenCheckingOnboardedService() {
     }
 
     @Test
-    void whenRetrievingSwagger() {
+    void whenRetrievingSwaggerUrl() {
         final String swaggerUrl = "https://hostname/sampleclient/api-doc";
         HashMap<String, String> metadata = new HashMap<>();
         metadata.put("apiml.apiInfo.api-v2.swaggerUrl", swaggerUrl);
-        assertEquals(swaggerUrl, verificationOnboardService.retrieveSwagger(metadata));
+        assertEquals(swaggerUrl, verificationOnboardService.findSwaggerUrl(metadata));
+    }
+
+    @Test
+    void whenRetrievingnNullSwaggerUrl() {
+        final String swaggerUrl = null;
+        HashMap<String, String> metadata = new HashMap<>();
+        metadata.put("apiml.apiInfo.api-v2.swaggerUrl", swaggerUrl);
+        assertEquals("", verificationOnboardService.findSwaggerUrl(metadata));
     }
 
 
     @Test
-    void whenRetrievingEmptySwagger() {
+    void whenRetrievingEmptySwaggerUrl() {
         HashMap<String, String> metadata = new HashMap<>();
         metadata.put("apiml.apiInfo.api-v2.swaggerUrl", null);
-        assertEquals("", verificationOnboardService.retrieveSwagger(metadata));
+        assertEquals("", verificationOnboardService.findSwaggerUrl(metadata));
+    }
+
+
+    @Nested
+    class givenEndpoint {
+        @Test
+        void whenEndpointReturnsDocumented400_thenReturnEmptyList() {
```

Comment:
do you have a test for a successful return code?

Reply:
Added one